test/scraped/response/decorator/absolute_urls_test.rb

```diff
@@ -107,4 +117,12 @@ def link(selector)
   it 'leaves empty link href attributes alone' do
     page.empty_link.must_equal ''
   end
+
+  it 'handles square brackets' do
+    page.bracketed_link.must_equal 'http://example.com/person[123]'
+  end
+
+  it 'encodes space characters' do
+    page.relative_link_with_unencoded_space.must_equal 'http://example.com/person 123'
```

Comment:
This test isn't doing what it says (or what it should). It claims that it's encoding space characters, but it's testing for 'http://example.com/person 123' rather than 'http://example.com/person%20123'

The resulting URLs from the decorator need to be things that `open-uri` can handle, which it won't here:

``` 
URI::InvalidURIError: bad URI(is not URI?): http://example.com/person 123
```

NB: I note that it also doesn't handle the URL I added in my test (square brackets are complex, which are why they need to be handled separately!)  I had mistakenly assumed (largely due to Chrome not auto-converting such URLs) that the un-escaped version was OK, but it looks like you'll also need to adapt that test to return '/person%5B123%5D'), to make it ruby-URI safe.